### PR TITLE
docs: add database design doc

### PR DESCRIPTION
**Pull request by ramraut-osm**
Added database design doc

## Summary by CodeRabbit

- **Documentation**
    - Enhanced the database design document for the Transcript Summarization App with detailed tables and relationships for user files and job logs.

### CODE REVIEW

File: apps/api/docs/database-design.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# **Database Design**
+
```
Comment (xixas) [inline apps/api/docs/database-design.md:2]: Keep the table names as `jobs` and `files`

File: apps/api/docs/database-design.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## **Data Dictionary**
+
+### User Files
+
+| Attribute | Data Type | Not Null | Default | Description |
+| --- | --- | --- | --- | --- |
+| task_id | int(11) | True |     | Primary key, unique identifier for the uploaded record |
+| input_file_path | varchar(255) |     |     | File path of user uploaded file |
+| output_file_path | varchar(255) |     |     | File path of summarized file stored in server after successful processing |
```
Comment (xixas) [inline apps/api/docs/database-design.md:27]: This is incorrect.
files table will just have file details, it doesn't has any concept of input and output files.
1 record corresponds to a particular file.

File: apps/api/docs/database-design.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+| modified_by | Varchar(45) |     |     | Stores the details of Modifier of the record |
+
+### Job Logs
+
+| Attribute | Data Type | Not Null | Default | Description |
+| --- | --- | --- | --- | --- |
+| job_id | int(11) | True |     | Primary key, unique identifier for the uploaded record |
+| task_id | varchar(255) | True |     | Stores the ID value for related task_id |
+| job_status | enum | True |     | Stores the job status |
+| input_file | varchar(255) |     |     | File path of user uploaded file |
```
Comment (xixas) [inline apps/api/docs/database-design.md:42]: input and output file  should be foreign key to files table.
Actual details of file will be present in files table.

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/api/docs/database-design.md (hunk 1 above)
Comment (ramraut-osm) [reply apps/api/docs/database-design.md:2]: Table names changed to jobs and files.
 **files :** 
 **jobs :**

File: apps/api/docs/database-design.md (hunk 2 above)
Comment (ramraut-osm) [reply apps/api/docs/database-design.md]: Removed the input and output file path fields from files table.

**Review by xixas — APPROVED** [review]

**Pull request merged**